specs/SelectMulti.spec.ts

```diff
@@ -12,16 +12,80 @@ describe('SelectMulti', () => {
 	let wrapper: Wrapper<Vue>
 	let options: SelectOption[]
 
+	function mountSelectMulti(selectMultiTemplate: string) {
+		return mount({
+			data() { return { selectedOptions: [], options }},
+			template: `<div>${selectMultiTemplate}</div>`,
+			components: { SelectMulti },
+			methods: {
+				setOptions(newOptions) { this['options'] = newOptions }
+			}
+		})
+	}
+
+	function testBehaviorForSelectingMultipleOptions(originalOptions: SelectOption[], testBehaviorForEventEmitsAndVModel?: () => void) {
+		describe('when the second option is clicked & then the third option is clicked', () => {
+			beforeEach(async () => {
+				const listbox = wrapper.find("div[role=listbox]")
+				const divOptions = listbox.findAll('div[role=option]')
+				
+				await divOptions.at(1).trigger('click')
+				await divOptions.at(2).trigger('click')
+			})
+
+			it('displays the correctly selected pills', async () => {
+				const selectedOptionsText = wrapper.find('.selected-options').text()
+
+				expect(selectedOptionsText).toContain('Item Two')
+				expect(selectedOptionsText).toContain('Item Three')
+			})
+
+			it('marks the options in the dropdown as selected, with the correct class & aria attribute', async () => {
+				const listbox = wrapper.find("div[role=listbox]")
+				const selectedDivOptions = listbox.findAll('div.option-selected[role=option][aria-selected=true]')
+
+				expect(selectedDivOptions.at(0).text()).toEqual('Item Two')
+				expect(selectedDivOptions.at(1).text()).toEqual('Item Three')
+			})
+
+			testBehaviorForEventEmitsAndVModel?.()
+
+			describe('(testing finding items by value, rather than reference) ' + 
```

Comment:
This describe text is a little hard to follow - I'm following whats happening up until this point and then it I'm like 🤪 . But maybe thats just because this is a complicated use case? Thoughts?

Reply:
Yeah, let me take another shot at it. It's complicated, but it's important for it to make sense to future people